### ISSUE TITLE
cmd/vet: GoVersion=="" on standard library packages

### ISSUE THREAD

**Issue by timothy-king** [golang/go#64293]
Within standard library packages the GoVersion sent to unitchecker is "" for standard library packages. This can be recreated from `gotip vet -x ./...` from `$GOROOT/src` (which will be in a module).

This `GoVersion` is passed along to `go/types` in `types.Config`. When this is empty (which is an invalid version), `(*types.Info).FileVersions` is not populated. The `loopclosure` vet check uses `FileVersions` to decide what is the semantics of a given for loop is. So there will now be a divergence in behavior for cmd/vet when running on the standard libraries. 

The relevant code to set the field is in `buildVetConfig`:
```
	if a.Package.Module != nil {
		v := a.Package.Module.GoVersion
		if v == "" {
			v = gover.DefaultGoModVersion
		}
		vcfg.GoVersion = "go" + v
	}
```

A hunch for why `a.Package.Module` is nil is that it comes from `PackageModuleInfo` which documents that nil is expected:
```
// PackageModuleInfo returns information about the module that provides
// a given package. If modules are not enabled or if the package is in the
// standard library or if the package was not successfully loaded with
// LoadPackages or ImportFromFiles, nil is returned.
func PackageModuleInfo(ctx context.Context, pkgpath string) *modinfo.ModulePublic {
```

cc @bcmills

**Comment by bcmills** [comment]
@timothy-king, if you can write a regression test, I think I see the fix. Probably something like:
```go
	if a.Package.Module != nil {
		v := a.Package.Module.GoVersion
		if v == "" {
			v = gover.DefaultGoModVersion
		}
		vcfg.GoVersion = "go" + v
	} else if a.Package.Standard {
		vcfg.GoVersion = gover.LocalToolchain()
	}
```

**Comment by samthanawalla** [comment]
Addressed in #65612
Should be fixed now :)

GoVersion will be set to the local go version in this case.
